src/components/InterviewRejectReasonDropdown/InterviewRejectReasonDropdown.tsx

```diff
@@ -20,71 +9,68 @@ interface RejectInterviewStatusProps {
     rejectReasons: RejectReason[];
     onChangeRejectReasons?: (selected: string) => void;
 }
+
 export const InterviewRejectReasonDropdown = ({
     rejectReasons,
     onChangeRejectReasons,
 }: RejectInterviewStatusProps): JSX.Element => {
     const [isOpen, setIsOpen] = useState(false);
-    const standartOptions = rejectReasons?.map((option, index) => ({
-        text: option.text,
-        id: index,
-    }));
-    const [selected, setSelected] = useState(standartOptions[0].text);
+    const [search, setSearch] = useState<string>('');
+    const [selectedRejectReason, setSelectedRejectReason] = useState<string>();
 
-    const onStateClick = useCallback(
-        (option: RejectReason) => {
-            selected === option.text ? setSelected(standartOptions[0].text) : setSelected(option.text);
+    const items = useMemo(() => {
+        const result = rejectReasons.filter(({ text }) =>
+            text.toLocaleLowerCase().includes(search.toLocaleLowerCase()),
+        );
+        if (!result.length) {
+            result.push({
+                text: search,
+                id: -1,
+            });
+        }
+
+        return result.map((option, index) => ({ text: option.text, id: index.toString() }));
+    }, [rejectReasons, search]);
 
-            onChangeRejectReasons?.(option.text);
-            setIsOpen(false);
-        },
-        [onChangeRejectReasons, selected, standartOptions],
-    );
     return (
         <div className={s.DropdownRejectStatus}>
-            <Dropdown isOpen={isOpen} onClose={() => setIsOpen(false)}>
-                <DropdownTrigger
+            <Select
+                items={items}
+                isOpen={isOpen}
+                onChange={(item) => {
+                    const { text } = item[0];
+                    setSearch(text);
+                    setSelectedRejectReason(text);
+                    onChangeRejectReasons?.(text);
```

Comment:
we can remove `selectedRejectReason` (it is unnecessary state)
You can add `value` props to `<InterviewRejectReasonDropdown />` and path the text from parent component